Location button

benblack769 opened the pull request:
So I decided to rewrite the geolocation code.

It now operates in the following way:

The first time location of the device is found, the map centers on that location (or the center of the canyon if the location is not in the bounds).

If you are in the canyon, the map follows you until you touch the map.

When you click the center button, it zooms to your position.

It looks pretty sleek, too.

![locate](https://cloud.githubusercontent.com/assets/10835669/17453608/659cd660-5b33-11e6-95a0-ce9861054d4b.png)

benblack769 committed:
rough but center feature is working

benblack769 committed:
made interface how I like it

benblack769 committed:
added comments

benblack769 committed:
added location popup back in

benblack769 committed:
minor tweak to center icon

benblack769 commented:
Two things to check out:

https://github.com/domoritz/leaflet-locatecontrol

Add:

enableHighAccuracy:true

to map.location options

benblack769 committed:
set enableHighAccuracy:true for geolocation

benblack769 commented:
Thoughts on leaflet-locatecontrol.
1. The zoom to location icon is plain horrible.
   ![zoomicon](https://cloud.githubusercontent.com/assets/10835669/17452062/b112bf78-5b20-11e6-986b-0bbb8a9ca536.png)
2. It has some nice features I hadn't thought of. It is also sufficient for our purposes. The one problem I see is that the options for adding a custom icon seem really bizarre, and so I am not quite sure how to deal with it. But that was a problem anyways, so whatever.

In conclusion, yeah it seems like a nice thing to have in our code.

benblack769 committed:
locate lib added

benblack769 committed:
now functional

benblack769 committed:
added comments and cleanup for locationcontrol

benblack769 committed:
tweaked location graphics

benblack769 committed:
merge conflicts resolved

kamba16 commented:
The source code on github suggests using bower or the Mapbox or JsDelivr CDN rather than the github ones in the script and link tags. Should we do that instead?

I also get this error: 
<img width="408" alt="screen shot 2016-08-07 at 5 13 21 pm" src="https://cloud.githubusercontent.com/assets/17392296/17466422/88abd39e-5cc4-11e6-89eb-1045f1ddeb65.png">

and can't actually see the locate icon my computer.

I don't think that the icon is horrible but the crosshairs one above is better and more clear.

benblack769 committed:
fixed js library ordering

benblack769 committed:
fixed cdn issue for now

benblack769 commented:
You are right about the CDN thing. I changed that to the same solution we are using for photoswipe.

That should be fixed, I just loaded the libraries in the wrong order, and fixed them on the remote server, but never committed them.

benblack769 committed:
fixed cdn issue better

kamba16 commented:
2 things: 1) what are lines 64 and 65 doing in map.js? why is circleStyle defined twice?
`circleStyle:{},
        circleStyle: {
            color: '#136AEC',
            fillColor: '#136AEC',
            fillOpacity: 0.15,
            weight: 4,
            opacity: 0.5
        },`

2) We should get rid of this popup - or, if we keep it, it should say "you are within ___ meters OF this point" (I know that the leaflet default was "from", but that is grammatically incorrect)
<img width="311" alt="screen shot 2016-08-07 at 6 59 16 pm" src="https://cloud.githubusercontent.com/assets/17392296/17467290/5f7713c8-5cd1-11e6-9f2a-608d49eda0b5.png">

benblack769 committed:
fixed spelling error and code redundancy

benblack769 commented:
Fixed both problems.

kamba16 commented:
I'm fine merging this. Let's see what Rachel thinks.

rekennelly commented:
I don't love the not fully opacity circle, but I guess it will make sense on the canyon map vs on a street.

I like the icon, and I'm fine merging it.
